Source/DafnyCore/Resolver.cs

```diff
@@ -17661,7 +17669,21 @@ class DefaultValueSubstituter : Substituter {
         }
         var resultType = new InferredTypeProxy() { KeepConstraints = true };
         e.Type = new SeqType(resultType);
-        AddXConstraint(e.tok, "ContainerResult", e.Seq.Type, resultType, "type does not agree with element type of {0} (got {1})");
+        AddXConstraint(e.tok, "ContainerResult", e.Seq.Type, resultType, "multi-selection has type {0} which is incompatible with expected type {1}");
+      }
+    }
+
+    /// <summary>
+    /// Allows to lazily call ToString on the given (sequence) type's element type.
+    /// 
+    /// When adding type constraints, a sequence type's element type may not be resolved,
+    /// so attempting to eagerly call ToString on it (for use in an error message) results in a null reference.
+    /// By wrapping the sequence type in this object, the error handling logic will only call ToString
+    /// on the element type after it is resolved.
```

Comment:
Very nifty.  I would just suggest rephrasing the comment:
```suggestion
    /// When adding type constraints, a sequence type's element type (`AsSeqType.Arg`) may not be resolved,
    /// so attempting to eagerly read it (for use in an error message) results in a null reference.
    /// By wrapping the sequence type in this object, the error handling logic will only read `AsSeqType.Arg`
    /// on the element type after it is resolved.
```

Reply:
Yes, that's a better way of putting it. Thanks!